### PR TITLE
Update type declarations

### PR DESCRIPTION
**Pull request by tinom9**
### What was wrong?

Related to Issue #3042

### How was it fixed?

Updated the `web3.types` module.

### Todo:
- [x] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Wow

### CODE REVIEW

File: web3/types.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ class LogReceipt(TypedDict):
     address: ChecksumAddress
     blockHash: HexBytes
     blockNumber: BlockNumber
-    data: HexStr
+    data: HexBytes
     logIndex: int
     payload: HexBytes
```
Comment (tinom9) [inline web3/types.py]: Should `payload` and `topic` be removed?

Comment (pacrob) [reply web3/types.py]: I think you are correct. I can't find any reference to expected `payload` or `topic` fields, not sure where those came from. Go ahead and remove, thank you!

Comment (tinom9) [reply web3/types.py]: Done!

**Comment by pacrob** [comment]
Looks good, @tinom9 ! Thanks for catching those details. I've rebased and will merge once tests are passing.

**Pull request merged**